tests/test_georaster.py

```diff
@@ -105,14 +105,24 @@ def test_downsampling(self):
         assert r.data.shape == (3, 328, 400)
 
     def test_copy(self):
-
+        """
+        Test that the copy method works as expected. In particular
+        when copying r to r2:
+        - creates a new memory file
+        - if r.data is modified and r copied, the updated data is copied
+        - if r is copied, r.data changed, r2.data should be unchanged
+        """
+        # Open dataset, update data and make a copy
         r = gr.Raster(datasets.get_path("landsat_B4"))
+        r.data += 5
         r2 = r.copy()
 
```

Comment:
In addition, one could check:
```python
assert r is not r2
```
This will raise an error if it's the same object:
```python
a = [1, 2]
b = a

assert a is b  # This passes

c = a.copy()

assert a is not c   # This also passes
```

Reply:
Will also add this in `test_satimg` and `test_dem`

Reply:
I added a new test in test_satimg, hope you noticed